rules/import-index.js

```diff
@@ -0,0 +1,27 @@
+'use strict';
+const isImportingIndex = m => m.endsWith('/') || m.endsWith('index.js') || m.endsWith('index');
```

Comment:
I did mean `index` and `index.js`. They are valid package names (and they apparently do exist).
You can use https://github.com/npm/validate-npm-package-name to check the validity of npm package names.

For `@foo/index.js`, it is as you said.